components/brave_wallet_ui/page/reducers/page_reducer.ts

```diff
@@ -67,13 +70,15 @@ export const createPageReducer = (initialState: PageState) => {
     return newState
   })
 
-  reducer.on(Actions.walletSetupComplete, (state: PageState) => {
-    const newState: PageState = {
+  reducer.on(Actions.walletSetupComplete, (state: PageState, payload?: boolean): PageState => {
+    // complete setup unless explicitly halted
+    const setupStillInProgress = !payload
+
+    return {
       ...state,
-      setupStillInProgress: false
+      mnemonic: undefined,
+      setupStillInProgress
```

Comment:
We can now prevent onboarding completion after creating a wallet (when desired)

Reply:
The page "container" component has logic to restrict routes based off of if the wallet has been created yet. The old behavior would mark the setup process as complete when a wallet was created. This variable change would then change the accessible routes, blocking onboarding screen access, and redirecting to the portfolio screen. Now, I can optionally not mark the setup process as complete, so that the user can continue viewing onboarding screens for steps such as the verify seed phrase screen

Reply:
Ok, that's interesting and I want to look into this a bit more then just to make sure this couldn't be used to bypass the wallet by re-onboarding and overriding the password or something similar to that